CodeGeneration/Sources/SyntaxSupport/DeclNodes.swift

```diff
@@ -1016,21 +1085,25 @@ public let DECL_NODES: [Node] = [
     kind: .functionSignature,
     base: .syntax,
     nameForDiagnostics: "function signature",
+    documentation: "A function signature that defines the interface of the function.",
     children: [
       Child(
         name: "parameterClause",
         deprecatedName: "input",
-        kind: .node(kind: .functionParameterClause)
+        kind: .node(kind: .functionParameterClause),
+        documentation: "The parameters of the function."
       ),
       Child(
         name: "effectSpecifiers",
         kind: .node(kind: .functionEffectSpecifiers),
+        documentation: "The side effect indicators of the function. They help clarify the behavior of the function.",
```

Comment:
`async` and `throw` do more than just clarifying the behavior of the function, they actually change the behavior of a function. How about focussing on `async` and `throws` here? I think that’s a concept that most Swift developers understand even if they don’t know the formal term for it.


Something like

```suggestion
        documentation: "The effect indicators of the function, like `async` or `throws`.",
```